src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -189,7 +189,7 @@ private function initializeSubscribers(): void
                 $listener = $this->container->get($listener);
             }
             // throw new \InvalidArgumentException(sprintf('Using Doctrine subscriber "%s" is not allowed, declare it as a listener instead.', \is_object($listener) ? $listener::class : $listener));
-            trigger_deprecation('symfony/doctrine-bridge', '6.3', 'Using Doctrine subscribers as services is deprecated, declare listeners instead');
+            trigger_deprecation('symfony/doctrine-bridge', '6.3', 'Registering "%s" as a Doctrine subscriber is deprecated. Register it as a listener with the #[AsDoctrineListener] attribute instead.', \is_object($listener) ? get_debug_type($listener) : $listener);
```

Comment:
@stof would that work for you?
```suggestion
            trigger_deprecation('symfony/doctrine-bridge', '6.3', 'Registering "%s" as a Doctrine subscriber is deprecated. Register it as a listener instead, using e.g. the #[AsDoctrineListener] attribute.', \is_object($listener) ? get_debug_type($listener) : $listener);
```

Reply:
this is a bit better. But it still provides a wrong migration path for anything else than the ORM listeners or for non-autowired projects.

Reply:
@alamirault can you take this message? I don't have a better idea, and while @stof is right, I think it's fine to optimize for the end users (bundle authors need more skills anyway, and this can server as a hint for them)

Reply:
@nicolas-grekas I updated PR